core/src/main/java/org/jdbi/v3/core/mapper/reflect/internal/ImmutablesPropertiesFactory.java

```diff
@@ -164,9 +164,12 @@ private MethodHandle findBuilderSetter(final Class<?> builderClass, String name,
                     return MethodHandles.lookup().unreflect(m);
                 }
             }
-            final IllegalArgumentException iae = new IllegalArgumentException("Failed to find builder setter for property " + name + " of " + type);
+            final IllegalArgumentException iae = new IllegalArgumentException("Failed to find builder setter for property " + name + " on " + builderClass);
             failures.forEach(iae::addSuppressed);
-            throw iae;
+            return MethodHandles.dropArguments(
+                    MethodHandles.throwException(Object.class, IllegalArgumentException.class),
```

Comment:
This crazy contraption constructs a MethodHandle that drops two arguments (`this` and the set value) and instead throws the constructed `iae`.  This laziness means if we don't ever try to call a setter, its absence is not fatal.

Reply:
So we return a method handle that throws when called, instead of throwing eagerly?

Reply:
Exactly.